processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -98,287 +167,247 @@ public void start()
+        } else if (notice instanceof DropNotice) {
+          DropNotice dropNotice = (DropNotice) notice;
+          lookupsToDrop.add(dropNotice.lookupName);
+          lookupsToLoad.remove(dropNotice.lookupName);
+        } else {
+          throw new ISE("Unknown Notice type [%s].", notice.getClass().getName());
         }
       }
-    }
-    finally {
+
+      return new LookupsState(Maps.newHashMap(lookupMap), lookupsToLoad, lookupsToDrop);
+    } finally {
       startStopLock.readLock().unlock();
     }
-    return update;
   }
 
-  /**
-   * @param lookupName name of {@link LookupExtractorFactory} to delete from the reference registry.
-   *                   this function does call the cleaning method {@link LookupExtractorFactory#close()}
-   *
-   * @return true only if {@code lookupName} is removed and the lookup correctly stopped
-   */
-  public boolean remove(String lookupName)
+  private void takeSnapshot()
   {
-    try {
-      startStopLock.readLock().lockInterruptibly();
-    }
-    catch (InterruptedException e) {
-      throw Throwables.propagate(e);
-    }
-    try {
-      final LookupExtractorFactory lookupExtractorFactory = lookupMap.remove(lookupName);
-      if (lookupExtractorFactory != null) {
-        LOGGER.debug("Removed lookup [%s]", lookupName);
-        if (lookupSnapshotTaker != null) {
-          lookupSnapshotTaker.takeSnapshot(getAllAsList());
-        }
-        return lookupExtractorFactory.close();
+    if (lookupSnapshotTaker != null) {
+      startStopLock.readLock().lock();
+
+      List<LookupBean> lookups;
+      try {
+        lookups = Lists.newArrayList(
+            Collections2.transform(
+                lookupMap.entrySet(),
+                new Function<Map.Entry<String, LookupExtractorFactoryContainer>, LookupBean>()
+                {
+                  @Nullable
+                  @Override
+                  public LookupBean apply(
+                      @Nullable
+                      Map.Entry<String, LookupExtractorFactoryContainer> input
+                  )
+                  {
+                    final LookupBean lookupBean = new LookupBean(input.getKey(), null, input.getValue());
+                    return lookupBean;
+                  }
+                }
+            )
+        );
       }
+      finally {
+        startStopLock.readLock().unlock();
+      }
+
+      lookupSnapshotTaker.takeSnapshot(lookups);
     }
-    finally {
-      startStopLock.readLock().unlock();
-    }
-    return false;
   }
 
-  /**
-   * @param lookupName key to fetch the reference of the object {@link LookupExtractor}
-   *
-   * @return reference of {@link LookupExtractorFactory} that correspond the {@code lookupName} or null if absent
-   *
-   * @throws IllegalStateException if the {@link LookupReferencesManager} is closed or did not start yet
-   */
-  @Nullable
-  public LookupExtractorFactory get(String lookupName)
+  private void assertStarted()
```

Comment:
Guava's convention is that names of methods that always make checks start with check-, names of methods that do assertions start with verify-